### PR TITLE
Fix isolated component events after parent is recreated

### PR DESCRIPTION
**Pull request by chromakode**
Here's a proposed fix to #409.

The added test case fails on master and passes w/ this new change. I'm reusing `isolatedElements` as a way to check whether the element has been replaced by a new instance (when the parent is recreated, the create hook seems to run first and then the old node gets destroyed).

### CODE REVIEW

File: dom/test/browser/isolation.js [hunk 1]
```diff
@@ -715,6 +715,46 @@ describe('isolation', function () {
...
+    let dispose;
+    sources.DOM.select(':root').elements().skip(1).subscribe(function (root) {
+      setTimeout(() => {
+        const foo = root.querySelector('.foo');
+        if (!foo) return;
+        foo.click();
+      }, 0);
+    });
+    setTimeout(function(){
+      assert.strictEqual(clicksCount, 3);
```
Comment (staltz) [inline dom/test/browser/isolation.js]: Thanks for the PR. Help me understand something: why are we expecting it to be `3`? The foo element was clicked just once.

Comment (chromakode) [inline dom/test/browser/isolation.js]: Yeah, this is kind of tricky -- I was puzzled by it too. It's copied from the test case immediately above.

What's happening is that the `:root` `.elements()` stream is emitting an event each time the DOM is updated. The main `DOM` stream emits 3 times so we expect 3 clicks. I'd be happy to rewrite this in a more direct fashion if you'd like.

Comment (staltz) [inline dom/test/browser/isolation.js]: Ah, I see it now. How about you put `elements().skip(1).take(3)` just to make it clear the intent is to test for 3 elements?

Comment (chromakode) [inline dom/test/browser/isolation.js]: Good idea! Done.

**Comment by chromakode** [comment]
I rebased to fix the first commit message -- changed the classification from "style" to "refactor".

**Comment by staltz** [comment]
Great! Merged. Thanks a lot @chromakode

**Pull request merged**